### ISSUE TITLE
Use 3rd party priority queues

### ISSUE THREAD

**Issue by qnikst** [haskell-distributed/distributed-process-extras#12]
It may be better not to reimplement priority queues and use existing one instead like:
https://hackage.haskell.org/package/psqueues

**Comment by hyperthunk** [comment]
Yeah, I agree. I'll move to that library (or an equivalent) shortly.
